src/SquidNew.cc

```diff
@@ -23,5 +23,12 @@ void operator delete(void *address)
     xfree(address);
 }
 
+// Squid does not use C++14 yet, but this declaration avoids a
+// -Wsized-deallocation error when building with C++14 implicitly enabled.
+void operator delete(void *address, size_t)
+{
+    operator delete(address);
+}
```

Comment:
If you apply Amos' suggestion, please do not remove the source code comment that explains what is going on here. Adjust that comment if needed, but please keep the otherwise missing explanation for this exceptional code.

Reply:
It's lower case __cplusplus, but it works. Commit 5da7bb892 to be pushed with the rest of the batch later.

Reply:
@rousskov : sure